### ISSUE TITLE
Clarification for Pipes and Filters Lesson - Passing output to another command heading

### ISSUE THREAD

**Issue by slfriedman** [swcarpentry/shell-novice#1317]
In the Pipes and Filters episode ([https://swcarpentry.github.io/shell-novice/04-pipefilter/index.html](https://swcarpentry.github.io/shell-novice/04-pipefilter/index.html)), the pipe is introduced under the subheader “Passing output to another command”.  The text that currently introduces that section is as follows (emphasis mine):

> In our example of finding the file with the fewest lines, we are using two intermediate files lengths.txt and sorted-lengths.txt to store output. This is a **confusing** way to work because even once you understand what wc, sort, and head do, those intermediate files make it hard to follow what’s going on. We can make it **easier** to understand by running sort and head together:

I don’t think the current framing of the multi-step process being more “confusing” and the use of the pipe being “easier” is all that friendly for new learners. I would propose adjusting that paragraph as follows (changes in bold):

> In our example of finding the file with the fewest lines, we are using two intermediate files lengths.txt and sorted-lengths.txt to store output. **This method helped us to illustrate how each command worked, and build up to a more complex output, but it is not necessary to store the output in a file at each of these steps. In fact, there are some downsides to storing the output of every step in this way (extra files that will need to be discarded later, more code to type, possible confusion later if you are trying to retrace or replicate your work).
>It would be preferable if we could run our commands all in one go, without saving to a file in between. In the next step, we demonstrate how to do this by running sort and head together:**

**Comment by bkmgit** [comment]
@slfriedman Thanks for the suggestion. It is a little longer than the current version, can it be shortened? For example
> In our example of finding the file with the fewest lines, we use two intermediate files lengths.txt and 
> sorted-lengths.txt to store output.  Downsides of storing the output of every step include extra files 
> that will need to be discarded later, more code to type and possible confusion if you later try to 
> replicate your work.  In the next step, we run sort and head together without storing intermediate files:

**Comment by slfriedman** [comment]
@bkmgit That version looks fantastic, thank you for taking what I had and improving it further.

**Comment by bkmgit** [comment]
@slfriedman Pull requests and new contributors are welcome!
